sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/UnivocityParser.scala

```diff
@@ -300,14 +302,11 @@ private[csv] object UnivocityParser {
       lines
     }
 
-    val filteredLines: Iterator[String] =
-      CSVUtils.filterCommentAndEmpty(linesWithoutHeader, options)
```

Comment:
I feel sure I put this because Univocity has an issue about this before. Wouldn't we better just keep this just in case? I think we already do such things in Spark side redundantly to make sure.

Reply:
Seems fixed as the tests pass now. I hit a test failure _if I remember this correctly_.

Reply:
Probably, you observed issues in old versions of uniVocity parser as @maropu wrote above. I would propose to remove the filtering till we face to the cases when uniVocity's filter doesn't work as it is expected. So, we would submit an issue for uniVocity and revert the changes back.

> I think we already do such things in Spark side redundantly to make sure in few places.

I looked at another places where we do the same but this is only the place where we do filtering directly before uniVocity.

Reply:
I mean, for example, we do both Parquet's record-level filter and Spark's filter although they are pushed down.

Reply:
Also, IMO, even if these kinds of bugs fixed in libraries, we keep workaround for a while in most cases, e.g., https://issues.apache.org/jira/browse/SPARK-24109

Reply:
Guys, this is speculation about the bug, it seems to me. If the code was added to fix a bug in previous version fo uniVocity parser, where is the tests for that? For now, we filter CSV rows additionally to uniVocity only because we don't trust to the library as far as I understand @HyukjinKwon . Following to the logic, we should duplicate another functionality of uniVocity in CSV datasource. That looks pretty crazy.

Reply:
@MaxGekk, It doesn't always mean that we have tests. Because it was there from the first place and I tried to remove it, then the tests were broken. I expected to be broken again but seems passed now. So, I'm just guessing that it's fixed.

Usually we trust but we should be careful if there were some issues found. I don't think we should make this case special. I am not seeing meaningful improvement either.

One nit is, BTW, the purpose of `ignoreLeadingWhiteSpaceInRead` and `ignoreTrailingWhiteSpaceInRead` are basically for trimming the whitespaces in the values not for skipping empty lines.

Reply:
We need strong evidence/test cases to make sure the current uniVocity filtering works.

Reply:
I wrote a test in the PR: https://github.com/apache/spark/pull/21394 which is passed on the current implementation but fails on this PR. After this PR, lines with multiple whitespaces are not ignored. To ignore such lines, need to set `ignoreLeadingWhiteSpace` to `true`. See https://github.com/uniVocity/univocity-parsers/blob/v2.6.3/src/main/java/com/univocity/parsers/csv/CsvParser.java#L106-L110

Reply:
Oh, so now it actually fixes an issue, right? will take a look soon. BTW, I think you can fold the changes in #21394 into here.